gui/src/steamshortcutparser.cpp

```diff
@@ -0,0 +1,660 @@
+        std::string landscapeGridPath;
+        landscapeGridPath.append(gridPath);
+        landscapeGridPath.append(shortAppId);
+        landscapeGridPath.append(".png");
+        downloadFile(log, artwork["landscape"].c_str(), landscapeGridPath.c_str());
+
+        std::string portraitGridPath;
+        portraitGridPath.append(gridPath);
+        portraitGridPath.append(shortAppId);
+        portraitGridPath.append("p.png");
+        downloadFile(log, artwork["portrait"].c_str(), portraitGridPath.c_str());
+
+        std::string heroPath;
+        heroPath.append(gridPath);
+        heroPath.append(shortAppId);
+        heroPath.append("_hero.png");
+        downloadFile(log, artwork["hero"].c_str(), heroPath.c_str());
+
+        std::string logoPath;
+        logoPath.append(gridPath);
+        logoPath.append(shortAppId);
+        logoPath.append("_logo.png");
+        downloadFile(log, artwork["logo"].c_str(), logoPath.c_str());
+    }
+
+    uint64_t generatePreliminaryId(const std::string& exe, const std::string& appname) {
+        std::string key = exe + appname;
+
+        // Convert the CRC32 result to a BigInt
+        crc_t crcResult;
+
+        crcResult = crc_init();
+        crcResult = crc_update(crcResult, (unsigned char *) key.c_str(), strlen(key.c_str()));
+        crcResult = crc_finalize(crcResult);
+
+        uint64_t top = static_cast<uint64_t>(crcResult) | static_cast<uint64_t>(0x80000000);
+
+        // Perform bitwise operations
+        uint64_t preliminaryId = (top << 32) | static_cast<uint64_t>(0x02000000);
+
+        return preliminaryId;
+    }
+
+    // Function to generate App ID
+    std::string generateAppId(const std::string& exe, const std::string& appname) {
+        return std::to_string(generatePreliminaryId(exe, appname));
+    }
+
+    // Function to shorten App ID
+    std::string shortenAppId(const std::string& longId) {
+        return std::to_string(std::stoull(longId) >> 32);
+    }
+
+    // Function to generate Short App ID
+    std::string generateShortAppId(const std::string& exe, const std::string& appname) {
+        std::string longId = generateAppId(exe, appname);
+        return shortenAppId(longId);
+    }
+
+    uint32_t generateShortcutId(const std::string& exe, const std::string& appname) {
+        // Generate preliminary ID
+        uint64_t preliminaryId = generatePreliminaryId(exe, appname);
+
+        // Shift right by 32 bits and subtract 0x100000000
+        uint32_t shortcutId = static_cast<uint32_t>((preliminaryId >> 32) - 0x100000000);
+
+        return shortcutId;
+    }
+
+    std::string getValueFromMap(const std::map<std::string, std::string>& myMap, const std::string& key) {
+        auto it = myMap.find(key);
+        return (it != myMap.end()) ? it->second : ""; // Return the value or an empty string if the key is not found
+    }
+
+    bool containsString(const std::vector<std::string>& stringArray, const std::string& targetString) {
```

Comment:
Don't really need helper function for this. Also it's shorter if you use QString instead.